### PR TITLE
Reference disks in GCP Batch [WX-1819]

### PR DESCRIPTION
**Pull request by mcovarr**
### Description

Reference disks now working in the GCP Batch backend 🎉 

#### `CHANGELOG.md`
 - [x] I updated `CHANGELOG.md` in this PR
 - [ ] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [x] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

File: src/ci/resources/gcp_batch_shared_application.inc.conf [hunk 1]
```diff
@@ -30,5 +30,23 @@ backend {
...
+        include "gcp_batch_provider_config.inc.conf"
+
+        include "dockerhub_provider_config_v2.inc.conf"
+        # This SA does not have permission to bill this project when accessing RP buckets.
+        # This is on purpose so that we can assert the failure (see requester_pays_localization_negative)
+        batch.compute-service-account = "centaur@broad-dsde-cromwell-dev.iam.gserviceaccount.com"
+        filesystems.http {}
+
+        # Cromwell 57+ reference disk manifest specification included here.
+        include "papi_v2_reference_image_manifest.conf"
```
Comment (mcovarr) [inline src/ci/resources/gcp_batch_shared_application.inc.conf:48]: we'll want to rename this manifest file once we have things working

**Comment by mcovarr** [comment]
With the current code I see in the logs:

```
"docker: Error response from daemon: error while creating mount source path '/mnt/11a4324d4472f639f3fc558b00afeacd': mkdir /mnt/11a4324d4472f639f3fc558b00afeacd: read-only file system."
```

and in the job metadata the following. Note the lack of `ro` specifiers in the docker command line invocation, and the presence of `ro` specifiers in the `volumes`:

```
            "container": {
              "commands": [
                "-c",
                "printf '%s %s\\n' \"$(date -u '+%Y/%m/%d %H:%M:%S')\" Running\\ user\\ runnable:\\ docker\\ run\\ -v\\ /mnt/disks/cromwell_root:/mnt/disks/cromwell_root\\ -v\\ /mnt/11a4324d4472f639f3fc558b00afeacd:/mnt/11a4324d4472f639f3fc558b00afeacd\\ -v\\ /mnt/d9e025138b28caa42dd4006fc3636661:/mnt/d9e025138b28caa42dd4006fc3636661\\ --entrypoint\\=/bin/bash\\ ubuntu@sha256:8a37d68f4f73ebf3d4efafbcf66379bf3728902a8038616808f04e34a9ab63ee\\ /mnt/disks/cromwell_root/script"
              ],
              "entrypoint": "/bin/sh",
              "imageUri": "gcr.io/google.com/cloudsdktool/cloud-sdk:461.0.0-alpine",
              "volumes": [
                "/mnt/disks/cromwell_root:/mnt/disks/cromwell_root:rw",
                "/mnt/11a4324d4472f639f3fc558b00afeacd:/mnt/11a4324d4472f639f3fc558b00afeacd:ro",
                "/mnt/d9e025138b28caa42dd4006fc3636661:/mnt/d9e025138b28caa42dd4006fc3636661:ro"
              ]
            }
```

**Comment by dspeck1** [comment]
Thanks!  I was just about to ask you for the logs from GCP.  Looks like the issue is the mount path.  Batch only allows mounting to specific locations.  I will push a PR to update the mount path.

**Comment by mcovarr** [comment]
That did the trick for the one test currently in the PR! There are a few other PAPI v2 reference disk tests for which I'll make GCP Batch versions and add those here.

**Review by aednichols — APPROVED** [review]

**Review by jgainerdewar — APPROVED** [review]

**Comment by jbakerpmc** [comment]
Hello @mcovarr I'm trying to use the reference-disks option with GCP-Batch as my backend, I ran the `create_images.sh` (https://cromwell.readthedocs.io/en/develop/backends/GCPBatch/) to obtain my reference-disk-localization-manifests,. Also based on: https://cromwell.readthedocs.io/en/develop/backends/GCPBatch/However, I'm not sure if this is deprecated based on this current fix. This is an example of what I am trying to include in my cromwell configuration (cromwell.conf). 

`reference-disk-localization-manifests = [
  {
    "imageIdentifier" : "projects/private-test-cromwell/global/images/omics-reference-disk-image",
    "diskSizeGb" : 10,
    "files" : [ {
        "path" : "cromwell-references/hg38/v0/Homo_sapiens_assembly38.dict",
        "crc32c" : 2158779318
      }, {
        "path" : "cromwell-references/hg38/v0/Homo_sapiens_assembly38.fasta",
        "crc32c" : 420322484
      }, {
        "path" : "cromwell-references/hg38/v0/Homo_sapiens_assembly38.fasta.fai",
        "crc32c" : 1970999569
      }
    ]
  } ]`

**Comment by mcovarr** [comment]
@jbakerpmc there have been no changes to reference disk localization configuration between GCP Batch and PAPI v2 beta that I'm aware of.  You can `include` if you prefer to keep reference disk config in a separate file, or just inline to your main config if you prefer.

**Comment by jbakerpmc** [comment]
@mcovarr Thanks for your response. The documentation can be somewhat unclear. I’m currently working with Cromwell-87 and have also tried a custom build from the release 88 dev branch. While running a job, I’m encountering issues when attempting to mount my files to a local mount. For instance:

```
severity: "ERROR"
textPayload: "Copying gs://test-cromwell-genomics-resources/references/hg38/v0/Homo_sapiens_assembly38.fasta.fai to file:///mnt/disks/cromwell_root/test-cromwell-genomics-resources/references/hg38/v0/Homo_sapiens_assembly38.fasta.fai"
```

**Comment by mcovarr** [comment]
@jbakerpmc GCP Batch reference disks are broken in Cromwell 87, you'll need to run from the `develop` branch at least until Cromwell 88 is released.

**Comment by jbakerpmc** [comment]
@mcovarr I continue to get the same error after creating a build with the develop branch seems that it is not ready! Thanks for letting me know :)

**Pull request merged**